lib/matrix.h

```diff
@@ -75,6 +75,15 @@ inline void Matrix2Multiply(const Array1& u, Array2& matrix) {
   }
 }
 
+// Dagger a 2x2 matrix.
+template <typename Array>
+inline void Matrix2Dagger(Array& matrix) {
+  for(int i = 1;i < 8;i+=2) {
```

Comment:
Fix spaces and use unsigned for consistency: `for (unsigned i = 1; i < 8; i += 2) {`.